src/Symfony/Component/DependencyInjection/Attribute/AutowireLocator.php

```diff
@@ -11,32 +11,36 @@
 
 namespace Symfony\Component\DependencyInjection\Attribute;
 
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
 use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
-use Symfony\Component\DependencyInjection\ContainerInterface;
-use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Contracts\Service\Attribute\SubscribedService;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
 
 #[\Attribute(\Attribute::TARGET_PARAMETER)]
 class AutowireLocator extends Autowire
 {
-    public function __construct(string ...$serviceIds)
-    {
-        $values = [];
-
-        foreach ($serviceIds as $key => $serviceId) {
-            if ($nullable = str_starts_with($serviceId, '?')) {
-                $serviceId = substr($serviceId, 1);
-            }
-
-            if (is_numeric($key)) {
-                $key = $serviceId;
-            }
-
-            $values[$key] = new Reference(
-                $serviceId,
-                $nullable ? ContainerInterface::IGNORE_ON_INVALID_REFERENCE : ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE,
-            );
+    /**
+     * @see ServiceSubscriberInterface::getSubscribedServices()
+     *
+     * @param array<string|SubscribedService>|string $services An explicit list of services or a tag name
```

Comment:
for locators, don't we expect an array with string keys ? If yes, this should be documented in the type.

Reply:
It's fine to use "no key", then we use the type as the key, FQCN ids FTW